frontend/components/GitHubLink.jsx

```diff
@@ -15,22 +15,43 @@ const GitHubLink = ({
 
   if (branch) {
     href = `${baseHref}/tree/${encodeURIComponent(branch)}`;
-    title = 'View branch';
+    title = 'View branch on GitHub';
   } else if (sha) {
     href = `${baseHref}/commit/${sha}`;
-    title = 'View commit';
+    title = 'View commit on GitHub';
+  }
+  function chooseIcon(iconStr) {
+    switch (iconStr) {
+      case 'branch':
+        return (
+          <IconBranch />
+        );
+      case 'commit':
+      case 'sha':
+        // return (
+        //   <IconBranch />
+        // )
+        break;
+      case 'repo':
+      default:
+        return (
+          <IconGitHub />
+        );
+    }
+    return iconStr;
   }
 
   return (
     <a
-      className="repo-link"
+      className={`${icon}-link`}
       href={href}
       title={title}
       target="_blank"
       rel="noopener noreferrer"
     >
+
+      { (chooseIcon(icon))}
```

Comment:
This is a note for future consideration. We have no conventions or standards in place for how we structure components. Could this be a general icon component that handles the select logic within the component? ie the above could be`<Icon name={icon} />`. If we come to some convention, we could apply it to our common components with just a few props to differentiate them.

Reply:
I was kind of following the convention here:
https://github.com/cloud-gov/pages-core/blob/348e91b6483f191904f2e78082165530f1f8b924/frontend/components/site/siteBuilds.jsx#L24

I do like the idea of cleaner component structure though.

Reply:
Started tracking the creation of a future standard in https://github.com/cloud-gov/pages-core/issues/4332